v1/charm-dev.yaml

```diff
@@ -88,33 +97,57 @@ instances:
           lxd init --auto
           adduser ubuntu lxd
 
+        - |
+          # Make sure juju directory is there
+          # https://bugs.launchpad.net/juju/+bug/1995697
+          sudo -u ubuntu mkdir -p /home/ubuntu/.local/share/juju
+
         - |
           # setup microk8s and bootstrap
-          adduser ubuntu microk8s
+          usermod -a -G snap_microk8s ubuntu
+          usermod -a -G microk8s ubuntu
           microk8s status --wait-ready
-          
+
           microk8s.enable metrics-server
           microk8s.kubectl rollout status deployments/metrics-server -n kube-system -w --timeout=600s
 
           # The dns addon will restart the api server so you may see a blip in the availability
-          # Separating addons to avoid errors such as 
+          # Separating addons to avoid errors such as
           # dial tcp 127.0.0.1:16443: connect: connection refused
-          microk8s.enable dns
+          microk8s.enable dns rbac
           microk8s.kubectl rollout status deployments/coredns -n kube-system -w --timeout=600s
-          
+
+          # wait for storage become available
           microk8s.enable hostpath-storage
-          microk8s.enable ingress
-          # wait for storage, ingress to become available
+          #microk8s.enable ingress
           microk8s.kubectl rollout status deployments/hostpath-provisioner -n kube-system -w --timeout=600s
-          microk8s.kubectl rollout status daemonsets/nginx-ingress-microk8s-controller -n ingress -w --timeout=600s
+          #microk8s.kubectl rollout status daemonsets/nginx-ingress-microk8s-controller -n ingress -w --timeout=600s
 
-          sudo -u ubuntu juju bootstrap --no-gui microk8s charm-dev
-          sudo -u ubuntu juju add-model --config logging-config="<root>=WARNING; unit=DEBUG" --config update-status-hook-interval="60m" welcome
+          # MetalLB
+          IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')
+          microk8s enable metallb:$IPADDR-$IPADDR
+          microk8s.kubectl rollout status daemonset.apps/speaker -n metallb-system -w --timeout=600s
+
+          ## Manually copy credentials to /var/snap/juju/current/microk8s/credentials/client.config
+          #mkdir -p /var/snap/juju/current/microk8s/credentials
+          #microk8s config | tee /var/snap/juju/current/microk8s/credentials/client.config > /dev/null
+
+          # For some reason, without this sleep, bootstrapping a controller errors out with "Terminated".
+          sleep 30
```

Comment:
Since this is here, I'm guessing there is no way to poll something to better tell when things are ready for this.  30 seconds seems arbitrary and is meh, but if all other avenues to do this better are exhausted, I guess this will have to do,

Reply:
Yep, haven't figured out a different way yet.

Reply:
In my testing, I removed this 30 second sleep and I don't see the error.  I don't suppose there is a way to get extra verbosity out of the `bootstrap` command to get more info on why it terminated?  I'm really not a fan of this sleep since it is already taking quite a long time for this to run and adding 30 seconds here only prolongs the poor user experience.  I'd really like to understand the problem first before just accepting this.

Reply:
Nice, thanks!